File: ring-android/libringclient/src/main/java/cx/ring/utils/SwigNativeConverter.java
/*
 *  Copyright (C) 2004-2020 Savoir-faire Linux Inc.
 *
 *  Author: Alexandre Lision <alexandre.lision@savoirfairelinux.com>
 *
 *  This program is free software; you can redistribute it and/or modify
 *  it under the terms of the GNU General Public License as published by
 *  the Free Software Foundation; either version 3 of the License, or
 *  (at your option) any later version.
 *
 *  This program is distributed in the hope that it will be useful,
 *  but WITHOUT ANY WARRANTY; without even the implied warranty of
 *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 *  GNU General Public License for more details.
 *
 *  You should have received a copy of the GNU General Public License
 *  along with this program; if not, write to the Free Software
 *  Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
 */

package cx.ring.utils;

import java.util.ArrayList;
import java.util.List;
import java.util.Map;

import cx.ring.daemon.Message;
import cx.ring.daemon.MessageVect;
import cx.ring.daemon.StringMap;
import cx.ring.daemon.StringVect;
import cx.ring.daemon.VectMap;

public class SwigNativeConverter {

    public static VectMap toSwig(List<Map<String, String>> creds) {
        VectMap toReturn = new VectMap();
        toReturn.reserve(creds.size());
        for (Map<String, String> aTodecode : creds) {
            toReturn.add(StringMap.toSwig(aTodecode));
        }
        return toReturn;
    }

    public static ArrayList<String> toJava(StringVect vector) {
        return new ArrayList<>(vector);
    }

    public static ArrayList<Message> toJava(MessageVect vector) {
        int size = vector.size();
        ArrayList<Message> toReturn = new ArrayList<>(size);
        for (int i = 0; i < size; i++)
            toReturn.add(vector.get(i));
        return toReturn;
    }

}

File: ring-android/libjamiclient/src/main/kotlin/net/jami/utils/SwigNativeConverter.kt
/*
 *  Copyright (C) 2004-2022 Savoir-faire Linux Inc.
 *
 *  Author: Alexandre Lision <alexandre.lision@savoirfairelinux.com>
 *
 *  This program is free software; you can redistribute it and/or modify
 *  it under the terms of the GNU General Public License as published by
 *  the Free Software Foundation; either version 3 of the License, or
 *  (at your option) any later version.
 *
 *  This program is distributed in the hope that it will be useful,
 *  but WITHOUT ANY WARRANTY; without even the implied warranty of
 *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 *  GNU General Public License for more details.
 *
 *  You should have received a copy of the GNU General Public License
 *  along with this program; if not, write to the Free Software
 *  Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
 */
package net.jami.utils

import net.jami.daemon.*
import java.util.ArrayList

object SwigNativeConverter {
    fun toSwig(creds: List<Map<String, String>>): VectMap =
        creds.mapTo(VectMap().apply { reserve(creds.size.toLong()) }) {
            StringMap.toSwig(it)
        }

    fun toJava(vector: StringVect): ArrayList<String> = ArrayList(vector)

    fun toJava(vector: MessageVect): ArrayList<Message> {
        val size = vector.size
        val toReturn = ArrayList<Message>(size)
        for (i in 0 until size) toReturn.add(vector[i])
        return toReturn
    }
}